homeassistant/components/motion_blinds/cover.py

```diff
@@ -0,0 +1,259 @@
+        if ATTR_POSITION in kwargs:
+            position = kwargs[ATTR_POSITION]
+            self._blind.Set_position(100 - position)
+
+    def stop_cover(self, **kwargs):
+        """Stop the cover."""
+        self._blind.Stop()
+
+
+class MotionTiltDevice(MotionPositionDevice):
+    """Representation of a Motion Blind Device."""
+
+    @property
+    def current_cover_tilt_position(self):
+        """
+        Return current angle of cover.
+
+        None is unknown, 0 is closed/minimum tilt, 100 is fully open/maximum tilt.
+        """
+        if self._blind.angle is None:
+            return None
+        return self._blind.angle * 100 / 180
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        self._blind.Set_angle(180)
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        self._blind.Set_angle(0)
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        if ATTR_TILT_POSITION in kwargs:
+            angle = kwargs[ATTR_TILT_POSITION] * 180 / 100
+            self._blind.Set_angle(angle)
+
+    def stop_cover_tilt(self, **kwargs):
+        """Stop the cover."""
+        self._blind.Stop()
+
+
+class MotionTDBUDevice(MotionPositionDevice):
+    """Representation of a Motion Top Down Bottom Up blind Device."""
+
+    def __init__(self, coordinator, blind, device_class, config_entry, motor):
+        """Initialize the blind."""
+        super().__init__(coordinator, blind, device_class, config_entry)
+        self._motor = motor
+        self._motor_key = motor[0]
+
+        if self._motor not in ["Bottom", "Top"]:
+            _LOGGER.error("Unknown motor '%s'", self._motor)
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._blind.mac}-{self._motor}"
+
+    @property
+    def name(self):
+        """Return the name of the blind."""
+        return f"{self._blind.blind_type}-{self._motor}-{self._blind.mac}"
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is open, 100 is closed.
+        """
+        if self._blind.position is None:
+            return None
+
+        return 100 - self._blind.position[self._motor_key]
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed or not."""
+        if self._blind.position is None:
+            return None
+
+        return self._blind.position[self._motor_key] == 100
+
+    def open_cover(self, **kwargs):
+        """Open the cover."""
+        self._blind.Open(motor=self._motor_key)
+
+    def close_cover(self, **kwargs):
+        """Close cover."""
+        self._blind.Close(motor=self._motor_key)
+
+    def set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        if ATTR_POSITION in kwargs:
```

Comment:
`ATTR_POSITION` is required

https://www.home-assistant.io/integrations/cover/#service-coverset_cover_position

Reply:
done